template/androidApp/src/main/java/io/bloco/template/MainActivity.kt

```diff
@@ -18,8 +37,50 @@ class MainActivity : ComponentActivity() {
     @InstallIn(ActivityComponent::class)
     interface ViewModelFactoryProvider : BaseViewModelFactoryProvider
 
+    private val viewModel: MainActivityViewModel by viewModels()
+
     override fun onCreate(savedInstanceState: Bundle?) {
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
-        setContent { TemplateTheme { TemplateNaveHost() } }
+
+        var uiState: MainActivityUiState by mutableStateOf(Loading)
+
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.uiState
+                    .onEach {
+                        uiState = it
+                    }.collect()
+            }
+        }
+
+        // Keep the splash screen on-screen until the UI state is loaded. This condition is
+        // evaluated each time the app needs to be redrawn so it should be fast to avoid blocking
+        // the UI.
+        splashScreen.setKeepOnScreenCondition {
+            when (uiState) {
+                Loading -> true
+                Success -> false
+            }
+        }
+
+        // Turn off the decor fitting system windows, which allows us to handle insets,
+        // including IME animations
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+
+        setContent {
+            val systemUiController = rememberSystemUiController()
+            val darkTheme = isSystemInDarkTheme()
+
+            // Update the dark content of the system bars to match the theme
+            DisposableEffect(systemUiController, darkTheme) {
+                systemUiController.setSystemBarsColor(if (darkTheme) Color.Transparent else Color.White)
+                systemUiController.setStatusBarColor(if (darkTheme) Color.Transparent else Color.White)
+                systemUiController.systemBarsDarkContentEnabled = !darkTheme
+                onDispose {}
```

Comment:
Is this needed?

Reply:
I think so, that's how was made in [NIA](https://github.com/android/nowinandroid/blob/639402dfb02eb7a7b45938bf2039a2d8c3a2b654/app/src/main/java/com/google/samples/apps/nowinandroid/MainActivity.kt#L102).